modules/EvseV2G/connection.cpp

```diff
@@ -673,6 +675,10 @@ static void* connection_handle_tcp(void* data) {
     if (shutdown(conn->conn.socket_fd, SHUT_RDWR) == -1) {
         dlog(DLOG_LEVEL_ERROR, "shutdown() failed: %s", strerror(errno));
     }
+
+    // Waiting for client closing the connection
+    std::this_thread::sleep_for(std::chrono::seconds(5));
```

Comment:
We tested this a lot in Cleveland. I think we should also wait before we send the shutdown and not only the close. Some cars did not really like that we send a FIN first even if we keep the connection running. They insisted that the EV should send the first FIN. Apart from that it is good.

Reply:
Good point, then I'll add a sleep of 2 seconds before shutdown. And change the 5 seconds sleep after shutdown to 3 seconds

Reply:
And I'll also add a sleep before the TLS shutdown. I also suspect that it is shutting down too quickly.